app/scripts/modules/core/src/application/application.state.provider.ts

```diff
@@ -33,6 +33,19 @@ export class ApplicationStateProvider implements IServiceProvider {
     this.childStates.push(this.insightState);
   }
 
+  private static getDefaultState(app: Application) {
+    const environments = app.getDataSource('environments');
+    const serverGroups = app.getDataSource('serverGroups');
+
+    if (environments && !environments.disabled) {
+      return '.environments';
+    } else if (serverGroups && !serverGroups.disabled) {
+      return '.insight.clusters';
+    } else {
+      return app.dataSources.find((ds) => ds.sref && !ds.disabled).sref;
+    }
```

Comment:
I was going to suggest adding 'pipelines' to this list, then I found this little guy: https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/application/service/ApplicationDataSourceRegistry.ts#L6-L13

What would you think of 
```
const defaultDataSource = ApplicationDataSourceRegistry.getDataSources()
  .find(x => !x.disabled && x.visible && x.sref);
return defaultDataSource?.sref || 'home.search';
```
?

Reply:
Note this would have the effect of making 'executions' higher priority than 'clusters', so we might want to swap those in the `defaultDataSourceOrder` if we took this approach?

Reply:
One more comment: the appealing thing about this approach to me is that it seems like a reasonable source of truth for default data source.

Reply:
YES this is much nicer, seems obvious in hindsight (especially since I just touched that default data source order thing the other day). I've updated the PR. 

The only thing I adjusted is that I'm still using `app.dataSources` instead of `ApplicationDataSourceRegistry.getDataSources()`, because `app.dataSources` [respects the enabled/disabled flags](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/application/service/ApplicationReader.ts#L71-L101) that come back from front50 whereas the data source configs from the registry are totally static. They both use the same ordering logic from the registry.

Reply:
> app.dataSources respects the enabled/disabled flags that come back from front50 whereas the data source configs from the registry are totally static.

😕